sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Distinct.java

```diff
@@ -96,8 +108,28 @@
               public Void apply(Iterable<Void> iter) {
                 return null; // ignore input
                 }
-            }))
-        .apply(Keys.<T>create());
+            }));
+    if (!acrossPanes) {
+      return combined.apply(Keys.<T>create());
```

Comment:
I think it would be enough to simply put a name in the apply - that should be the same thing, right?